samples: pmic: update npm1300_one_button sample to support nPM1304 (#23594)
nrfconnect/sdk-nrf: samples/pmic/native/npm13xx_one_button/README.rst

.. _npm13xx_one_button:

> nPM13xx: One button

Reviewer: `nPM13xx` is not allowed as exception on https://nordicsemi.atlassian.net/wiki/spaces/MAR/pages/132940950/Product+names#nPM-Family.
Therefore:
```suggestion
nPM1300 and nPM1304: One button
```

> ###################

Reviewer 2: header mismatch

.. table-from-sample-yaml::

> The sample also requires an `nPM1300 EK <nPM1300 EK product page_>`_ or an nPM1304 EK that you need to connect to the development kit as described in `Wiring`_.

Reviewer: Link to product page missing.

Reviewer: You've added it to links.txt in this PR. links.txt is not parsed by CI like :ref:s
```suggestion
The sample also requires an `nPM1300 EK <nPM1300 EK product page_>`_ or an `nPM1304 EK <nPM1304 EK product page_>`_ that you need to connect to the development kit as described in `Wiring`_.
```

   ============  ============  =========================================================
   < 1 sec       Short press   Sample flashes **LED** at 5 Hz, LDSW 1 enabled
   1 - 5 sec     Medium press  Sample flashes **LED** at 1 Hz, LDSW 1 disabled
   5 - 10 sec    Long press    PMIC enters ship mode provided **J3** is not connected
   > 10 sec      None          PMIC long press reset activates
   ============  ============  =========================================================

Ship mode is the lowest power state of the PMIC.
To exit ship mode, press the **SHPHLD** button or attach a USB cable to **J3**.

> In a production design, this will power off any device that is powered from an PMIC output.

Reviewer 3: ```suggestion
In a production design, this powers off any device that is powered from a PMIC output.
```

The long press reset performs a full power cycle of the PMIC and resets all setting to powerup defaults.

> In a production design, this will power cycle any device that is powered from an PMIC output.

Reviewer 3: ```suggestion
In a production design, this power cycles any device that is powered from an PMIC output.
```

> When using a separately powered development kit, you must restart the application to reconfigure the PMIC after exiting ship mode or long press reset.

Reviewer 3: ```suggestion
When using a separately powered development kit, you must restart the application to reconfigure the PMIC after exiting the ship mode or long press reset.
```
